apps/chat-e2e/src/ui/webElements/renameConversationModal.ts

```diff
@@ -0,0 +1,69 @@
+import { BaseElement } from './baseElement';
+
+import { isApiStorageType } from '@/src/hooks/global-setup';
+import { Attributes, Tags } from '@/src/ui/domData';
+import { keys } from '@/src/ui/keyboard';
+import { RenameConversationModalSelectors } from '@/src/ui/selectors';
+import { Page } from '@playwright/test';
+
+export class RenameConversationModal extends BaseElement {
+  constructor(page: Page) {
+    super(page, RenameConversationModalSelectors.modal);
+  }
+
+  public cancelButton = this.getChildElementBySelector(
+    RenameConversationModalSelectors.cancelButton,
+  );
+  public saveButton = this.getChildElementBySelector(
+    RenameConversationModalSelectors.saveButton,
+  );
+  public nameInput = this.getChildElementBySelector(Tags.input);
+  public title = this.getChildElementBySelector(
+    RenameConversationModalSelectors.title,
+  );
+
+  async editConversationNameWithSaveButton(
+    newName: string,
+    { isHttpMethodTriggered = true }: { isHttpMethodTriggered?: boolean } = {},
+  ) {
+    await this.nameInput.fillInInput(newName);
+    if (isApiStorageType && isHttpMethodTriggered) {
+      const respPromise = this.page.waitForResponse(
+        (resp) => resp.request().method() === 'DELETE',
+      );
+      await this.saveButton.click();
+      await respPromise;
+    } else {
+      await this.saveButton.click();
+    }
+  }
+
+  async editConversationNameWithEnter(
```

Comment:
this function duplicates the `editConversationNameWithSaveButton()`, the only difference is the trigger action. Let's intro a basic function with an action arg to avoid code duplication

Reply:
We had always the same approach before:

```typescript
  public async editConversationNameWithTick(
    newName: string,
    { isHttpMethodTriggered = true }: { isHttpMethodTriggered?: boolean } = {},
  ) {
    await this.openEditEntityNameMode(newName);
    const editInputActions = this.getEditInputActions();
    if (isApiStorageType && isHttpMethodTriggered) {
      const respPromise = this.page.waitForResponse(
        (resp) => resp.request().method() === 'DELETE',
      );
      await editInputActions.clickTickButton();
      return respPromise;
    }
    await editInputActions.clickTickButton();
  }

  public async editConversationNameWithEnter(newName: string) {
    await this.openEditEntityNameMode(newName);
    if (isApiStorageType) {
      const respPromise = this.page.waitForResponse(
        (resp) => resp.request().method() === 'DELETE',
      );
      await this.page.keyboard.press(keys.enter);
      return respPromise;
    }
    await this.page.keyboard.press(keys.enter);
  }
  ```

but done, anyway :)